extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -345,23 +342,22 @@ export {Scroller};
  * @param {!Object} classes
  * @return {PreactDef.Renderable}
  */
-function renderSlides(
-  {
-    alignment,
-    children,
-    loop,
-    mixedLength,
-    restingIndex,
-    offsetRef,
-    openLightbox,
-    pivotIndex,
-    snap,
-    snapBy,
-    visibleCount,
-    _thumbnails,
-  },
-  classes
-) {
+function renderSlides({
+  alignment,
+  children,
+  loop,
+  mixedLength,
+  restingIndex,
+  offsetRef,
+  openLightbox,
+  pivotIndex,
+  snap,
+  snapBy,
+  visibleCount,
+  _thumbnails,
+}) {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const classes = useStyles();
```

Comment:
@samouri 

This one's interesting:

1. We may need to forbid passing `useStyles()` around like an object.
2. Is `useStyles()` a hook in the way we're using it?

Reply:
I'm confused. Isn't calling `useStyles()` here invalid. useStyles is a hook, and you may only call hooks from React components (renderSlides is not a React component)

Reply:
One important note: it was an explicit goal that when adding the jss transform, it would be possible to run the code _without_ the transform and the code was still valid. I.e. it was simply an optimization for AMP mode, but not actually necessary for Preact mode. 

My understanding is that if we made changes like this, where non-React components call the hook, we would lose the ability to run untransformed code in preact mode.

Reply:
If we care about compatibility then we need to deopt in these cases. I've undone the source change, and updated the transform to fallback to the classes object.